keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/verification/VerificationRepository.kt

```diff
@@ -0,0 +1,33 @@
+package com.netflix.spinnaker.keel.api.verification
+
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.Verification
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import java.time.Instant
+
+interface VerificationRepository {
+  fun getState(
+    verification: Verification,
+    environment: Environment,
+    deliveryArtifact: DeliveryArtifact,
+    version: String
+  ) : VerificationState?
+
+  fun updateState(
+    verification: Verification,
+    environment: Environment,
+    deliveryArtifact: DeliveryArtifact,
+    version: String,
+    status: VerificationStatus
+  )
+}
+
+data class VerificationState(
+  val status: VerificationStatus,
+  val startedAt: Instant,
+  val endedAt: Instant?
+)
+
+enum class VerificationStatus(val complete: Boolean) {
+  RUNNING(false), PASSED(true), FAILED(true)
```

Comment:
We might want an `ERROR` state to model cases where something unexpected happened that prevented keel from carrying out the verification.